dpki/src/keystore.rs

```diff
@@ -0,0 +1,330 @@
+use crate::{
+    key_bundle::KeyBundle,
+    keypair::generate_random_sign_keypair,
+    seed::{generate_random_seed_buf, IndexedSeed, RootSeed, SeedContext, SeedTrait, SeedType},
+    utils, SEED_SIZE,
+};
+use holochain_core_types::{
+    agent::Base32,
+    cas::content::Address,
+    error::{HcResult, HolochainError},
+    signature::Signature,
+};
+
+use holochain_sodium::{kdf, secbuf::SecBuf};
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+
+pub enum Secret {
+    Key(KeyBundle),
+    RootSeed(RootSeed),
+    IndexedSeed(IndexedSeed),
+}
+
+#[allow(dead_code)]
+struct Keystore {
+    keys: HashMap<String, Arc<Mutex<Secret>>>,
+}
+
+impl Keystore {
+    #[allow(dead_code)]
+    pub fn new() -> Self {
+        Keystore {
+            keys: HashMap::new(),
+        }
+    }
+
+    /// return a list of the identifiers stored in the keystore
+    #[allow(dead_code)]
+    pub fn list(&self) -> Vec<String> {
+        self.keys.keys().map(|k| k.to_string()).collect()
+    }
+
+    /// adds a random root seed into the keystore
+    #[allow(dead_code)]
+    pub fn add_random_seed(&mut self, id_str: &str, size: usize) -> HcResult<()> {
+        let id = id_str.to_string();
+        if self.keys.contains_key(&id) {
+            return Err(HolochainError::ErrorGeneric(
+                "identifier already exists".to_string(),
+            ));
+        }
+        let seed_buf = generate_random_seed_buf(size);
+        let secret = Arc::new(Mutex::new(Secret::RootSeed(RootSeed::new(seed_buf))));
```

Comment:
How do we know this was intended to be a root seed? Perhaps the user just wishes to create a one-off application that doesn't follow the seed derivation (not sure why they'd do this...) but, I was imagining this store just being a map of `[u8; 32]`s that could have arbitrary meaning.

Reply:
Well as I said I had assumed that most of these functions were supposed to be wrappers of what had been written in the dpki crate, so I didn't know which else to use and just made the assumption.  Are you thinking that I should simplify this and not even use the seed types that were created in the crate?  And if so then move the derivations directly into the functions I implemented and not use those Seed type implementations in the Secret variants?

 (And if they aren't supposed to be wrappers of those functions, then who is actually using those functions?)

Reply:
removed now and there is just a direct buf stored as the seed.